### PR TITLE
go back to the v0.6.3 solution for VS2015 static_assert workaround

### PR DESCRIPTION
**Pull request by sezero**

### CODE REVIEW

**Comment by sezero** [comment]
Notes: There are the following _plain_ `static_assert`s that you did not
change with your VS2015 patch: Do they not need changing?
- Kss_Cpu.h at line 45
- Ay_Cpu.h at line 36
- Effects_Buffer.cpp at lines 35 and 39
- Fir_Resampler.h at line 81
- Gb_Cpu.cpp at line 117

File: gme/blargg_common.h [hunk 1]
```diff
@@ -27,10 +27,11 @@
 	#define STATIC_CAST(T,expr) ((T) (expr))
 #endif
 
-#if !defined(_MSC_VER) || _MSC_VER >= 1910
-	#define blaarg_static_assert(cond, msg) static_assert(cond, msg)
+#if defined(_MSC_VER) && _MSC_VER < 1910
+	#define blaarg_static_assert(cond, msg) \
+	        void blargg_failed_(int (*arg) [2 / (int) !!(cond) - 1])
```
Comment (Wohlstand) [inline gme/blargg_common.h:32]: I don't know, but I never liked the `!!` way to turn into the boolean, I prefer the explicit `(bool)` cast. Anyway, I bet, double-negation takes more instructions than the casting.

**Comment by Wohlstand** [comment]
> Notes: There are the following _plain_ `static_assert`s that you did not
> change with your VS2015 patch: Do they not need changing?

Actually they don't need. The reason of the workaround is because of special condition with the static assert that happens with MSVC 2015 and older in given places, but in other cases it just works. That why these static assets wasn't changed into workaround.

File: gme/blargg_common.h (hunk 1 above)
Comment (sezero) [reply gme/blargg_common.h:32]: It's a compile-time calculation  (and I actually like `!!` :)

**Comment by sezero** [comment]
> > Notes: There are the following _plain_ `static_assert`s that you did not
> > change with your VS2015 patch: Do they not need changing?
> 
> Actually they don't need. The reason of the workaround is because of special condition with the static assert that happens with MSVC 2015 and older in given places, but in other cases it just works. That why these static assets wasn't changed into workaround.

OK then

**Comment by Wohlstand** [comment]
Okay, I merged the thing and tweaked some at the end.

**Pull request merged**